src/Commands/ElasticsearchIndexCommand.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Rapidez\Core\Commands;
+
+use Carbon\Carbon;
+use Exception;
+use Rapidez\Core\Commands\InteractsWithElasticsearchCommand;
+
+abstract class ElasticsearchIndexCommand extends InteractsWithElasticsearchCommand
+{
+    /**
+     * Index all items in all stores
+     *
+     * @param  string $indexName
+     * @param  (callable(object): array)|array $data
+     * @param  (callable(object): array)|array $values
+     */
+    public function indexAll($indexName, $data, $values)
+    {
+        $stores = $this->getStores();
+        foreach ($stores as $store) {
+            $this->indexStore($store, $indexName, $data, $values);
+        }
+    }
+
+    /**
+     * Index all items in a specific stores
+     *
+     * @param  object $store
+     * @param  string $indexName
+     * @param  (callable(object): array)|array $data
+     * @param  (callable(object): array)|array $values
+     */
+    public function indexStore($store, $indexName, $data, $values)
+    {
+        $this->line('Indexing `'.$indexName.'` for store: '.$store->name);
+        try {
+            [$alias, $index] = $this->prepareIndexer($store, $indexName);
+
+            $currentData = is_callable($data)
+                ? $data(config()->get('rapidez.store_code'))
+                : $data;
+
+            foreach ($currentData as $item) {
+                $this->indexItem($index, $item, $item->id, $values);
+            }
+            $this->switchAlias($alias, $index);
+        } catch (Exception $e) {
+            $this->deleteIndex($index);
+            throw $e;
+        }
+    }
+
+    public function createAlias($store, $indexName): array
+    {
+        $alias = config('rapidez.es_prefix').'_'.$indexName.'_'.$store->store_id;
+        return [$alias, $alias.'_'.Carbon::now()->format('YmdHis')];
+    }
+
+    public function prepareIndexer($store, $indexName): array
+    {
+        $this->setStore($store);
+        [$alias, $index] = $this->createAlias($store, $indexName);
+        $this->createIndex($index);
+        return [$alias, $index];
+    }
+
+    public function indexItem($index, $item, $id, $values)
+    {
+        $currentValues = is_callable($values)
+            ? $values($item)
+            : collect($item)->only($values);
```

Comment:
If this is the only thing we do with the collection it may be more efficient to use the Arr::only() function which the collections only function uses as well https://github.com/laravel/framework/blob/82875a158e8f130a9103e096419f8518161b34d3/src/Illuminate/Collections/Collection.php#L912

Skips us having to put the array(able) into a collection first.